server/src/main/java/org/apache/druid/client/indexing/HttpIndexingServiceClient.java

```diff
@@ -350,7 +352,10 @@ public TaskPayloadResponse getTaskPayload(String taskId)
           )
       );
 
-      if (responseHolder.getContent().length() == 0) {
+      if (responseHolder.getContent().length() == 0 || responseHolder.getStatus() != HttpResponseStatus.OK) {
+        if (responseHolder.getStatus() == HttpResponseStatus.NOT_FOUND) {
+          log.info("Report not found for taskId [%s] because [%s]", taskId, responseHolder.getContent());
```

Comment:
Should we log if status is other non ok code too?

Reply:
Good idea..added logging for other non-ok statuses